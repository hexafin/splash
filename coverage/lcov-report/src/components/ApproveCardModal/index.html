<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/ApproveCardModal</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/components/ApproveCardModal
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">42.42% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>14/33</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">27.78% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>5/18</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">22.22% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/18</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">42.42% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>14/33</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="ApproveCardModal.js"><a href="ApproveCardModal.js.html">ApproveCardModal.js</a></td>
	<td data-value="42.42" class="pic low"><div class="chart"><div class="cover-fill" style="width: 42%;"></div><div class="cover-empty" style="width:58%;"></div></div></td>
	<td data-value="42.42" class="pct low">42.42%</td>
	<td data-value="33" class="abs low">14/33</td>
	<td data-value="27.78" class="pct low">27.78%</td>
	<td data-value="18" class="abs low">5/18</td>
	<td data-value="22.22" class="pct low">22.22%</td>
	<td data-value="18" class="abs low">4/18</td>
	<td data-value="42.42" class="pct low">42.42%</td>
	<td data-value="33" class="abs low">14/33</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu May 02 2019 23:21:01 GMT-0400 (Eastern Daylight Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
<script src="../../../block-navigation.js"></script>
</body>
</html>
