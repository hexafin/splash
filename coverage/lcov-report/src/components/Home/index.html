<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Home</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/components/Home
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">31.03% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>27/87</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">7.23% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>6/83</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">31.03% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>9/29</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">31.03% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>27/87</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="ColoredPayButton.js"><a href="ColoredPayButton.js.html">ColoredPayButton.js</a></td>
	<td data-value="60" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 60%;"></div><div class="cover-empty" style="width:40%;"></div></div></td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="10" class="abs medium">6/10</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="42.86" class="pct low">42.86%</td>
	<td data-value="7" class="abs low">3/7</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="10" class="abs medium">6/10</td>
	</tr>

<tr>
	<td class="file low" data-value="History.js"><a href="History.js.html">History.js</a></td>
	<td data-value="20.93" class="pic low"><div class="chart"><div class="cover-fill" style="width: 20%;"></div><div class="cover-empty" style="width:80%;"></div></div></td>
	<td data-value="20.93" class="pct low">20.93%</td>
	<td data-value="43" class="abs low">9/43</td>
	<td data-value="8.51" class="pct low">8.51%</td>
	<td data-value="47" class="abs low">4/47</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="10" class="abs low">2/10</td>
	<td data-value="20.93" class="pct low">20.93%</td>
	<td data-value="43" class="abs low">9/43</td>
	</tr>

<tr>
	<td class="file low" data-value="Home.js"><a href="Home.js.html">Home.js</a></td>
	<td data-value="32.14" class="pic low"><div class="chart"><div class="cover-fill" style="width: 32%;"></div><div class="cover-empty" style="width:68%;"></div></div></td>
	<td data-value="32.14" class="pct low">32.14%</td>
	<td data-value="28" class="abs low">9/28</td>
	<td data-value="3.13" class="pct low">3.13%</td>
	<td data-value="32" class="abs low">1/32</td>
	<td data-value="44.44" class="pct low">44.44%</td>
	<td data-value="9" class="abs low">4/9</td>
	<td data-value="32.14" class="pct low">32.14%</td>
	<td data-value="28" class="abs low">9/28</td>
	</tr>

<tr>
	<td class="file medium" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="6" class="abs medium">3/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="6" class="abs medium">3/6</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Apr 19 2019 16:14:10 GMT-0400 (Eastern Daylight Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
<script src="../../../block-navigation.js"></script>
</body>
</html>
