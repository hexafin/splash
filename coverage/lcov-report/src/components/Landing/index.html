<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Landing</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/components/Landing
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">63.27% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>31/49</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">41.67% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>15/36</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">46.15% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>6/13</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">61.7% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>29/47</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="Landing.js"><a href="Landing.js.html">Landing.js</a></td>
	<td data-value="31.82" class="pic low"><div class="chart"><div class="cover-fill" style="width: 31%;"></div><div class="cover-empty" style="width:69%;"></div></div></td>
	<td data-value="31.82" class="pct low">31.82%</td>
	<td data-value="22" class="abs low">7/22</td>
	<td data-value="22.73" class="pct low">22.73%</td>
	<td data-value="22" class="abs low">5/22</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	<td data-value="31.82" class="pct low">31.82%</td>
	<td data-value="22" class="abs low">7/22</td>
	</tr>

<tr>
	<td class="file high" data-value="LandingSwipeView.js"><a href="LandingSwipeView.js.html">LandingSwipeView.js</a></td>
	<td data-value="88.89" class="pic high"><div class="chart"><div class="cover-fill" style="width: 88%;"></div><div class="cover-empty" style="width:12%;"></div></div></td>
	<td data-value="88.89" class="pct high">88.89%</td>
	<td data-value="27" class="abs high">24/27</td>
	<td data-value="71.43" class="pct medium">71.43%</td>
	<td data-value="14" class="abs medium">10/14</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	<td data-value="88" class="pct high">88%</td>
	<td data-value="25" class="abs high">22/25</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Apr 19 2019 16:14:10 GMT-0400 (Eastern Daylight Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
<script src="../../../block-navigation.js"></script>
</body>
</html>
