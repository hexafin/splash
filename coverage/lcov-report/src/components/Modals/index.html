<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Modals</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/components/Modals
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">21.25% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>17/80</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">5.26% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/38</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">12.5% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/32</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">21.52% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>17/79</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="CardModal.js"><a href="CardModal.js.html">CardModal.js</a></td>
	<td data-value="10.87" class="pic low"><div class="chart"><div class="cover-fill" style="width: 10%;"></div><div class="cover-empty" style="width:90%;"></div></div></td>
	<td data-value="10.87" class="pct low">10.87%</td>
	<td data-value="46" class="abs low">5/46</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="24" class="abs low">0/24</td>
	<td data-value="5.56" class="pct low">5.56%</td>
	<td data-value="18" class="abs low">1/18</td>
	<td data-value="10.87" class="pct low">10.87%</td>
	<td data-value="46" class="abs low">5/46</td>
	</tr>

<tr>
	<td class="file medium" data-value="InfoModal.js"><a href="InfoModal.js.html">InfoModal.js</a></td>
	<td data-value="60" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 60%;"></div><div class="cover-empty" style="width:40%;"></div></div></td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="3" class="abs low">1/3</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	</tr>

<tr>
	<td class="file medium" data-value="ModalRoot.js"><a href="ModalRoot.js.html">ModalRoot.js</a></td>
	<td data-value="60" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 60%;"></div><div class="cover-empty" style="width:40%;"></div></div></td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="10" class="abs medium">6/10</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="3" class="abs low">1/3</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="9" class="abs medium">6/9</td>
	</tr>

<tr>
	<td class="file low" data-value="RaiseModal.js"><a href="RaiseModal.js.html">RaiseModal.js</a></td>
	<td data-value="15.79" class="pic low"><div class="chart"><div class="cover-fill" style="width: 15%;"></div><div class="cover-empty" style="width:85%;"></div></div></td>
	<td data-value="15.79" class="pct low">15.79%</td>
	<td data-value="19" class="abs low">3/19</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="11.11" class="pct low">11.11%</td>
	<td data-value="9" class="abs low">1/9</td>
	<td data-value="15.79" class="pct low">15.79%</td>
	<td data-value="19" class="abs low">3/19</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Apr 17 2019 19:32:05 GMT-0400 (Eastern Daylight Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
<script src="../../../block-navigation.js"></script>
</body>
</html>
