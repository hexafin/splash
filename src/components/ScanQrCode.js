import React, { Component } from "react"
import {
	View,
	Text,
	Image,
	Dimensions,
	StyleSheet,
	Alert
} from "react-native"
import { colors } from "../lib/colors"
import { defaults, icons } from "../lib/styles"
import { isIphoneX } from "react-native-iphone-x-helper"
import { connect } from "react-redux";
import { bindActionCreators } from "redux"
import { captureQr } from "../redux/transactions/actions"
import QRCodeScanner from 'react-native-qrcode-scanner'
import CloseButton from "./universal/CloseButton"
import Button from "./universal/Button"

const SCREEN_WIDTH = Dimensions.get("window").width
const SCREEN_HEIGHT = Dimensions.get("window").height

class ScanQrCode extends Component {
	render() {
		return (
			<QRCodeScanner
				ref={(node) => { this.scanner = node }}
				onRead={e => {
					try {
						const network = (this.props.network == 'testnet') ? bitcoin.networks.testnet : bitcoin.networks.bitcoin
						bitcoin.address.toOutputScript(address, network)
						this.props.captureQr(e.data)
						this.props.navigation.goBack(null)
					}
					catch (error) {
						Alert.alert("Invalid bitcoin address", null, [
							{text: "Dismiss", onPress: () => {
								this.scanner.reactivate()
							}}
						])
					}
				}}
				containerStyle={styles.container}
				topContent={
					<View style={styles.top}>
						<Image source={icons.qrIcon} resizeMode="contain" style={styles.icon}/>
						<Text style={styles.title}>Scan QR code</Text>
						<CloseButton onPress={() => {
							this.props.navigation.goBack(null)
						}}/>
					</View>
				}
				bottomContent={
					<View style={styles.bottom}>
						<Button title="Cancel" onPress={() => {
							this.props.navigation.goBack(null)
						}}/>
					</View>
				}/>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		height: SCREEN_HEIGHT,
		width: SCREEN_WIDTH,
		backgroundColor: colors.white
	},
	top: {
		flex: 1,
		width: SCREEN_WIDTH,
		backgroundColor: colors.primary,
		justifyContent: "center",
		alignItems: "center"
	},
	icon: {
		width: 40,
		height: 40,
		margin: 20,
	},
	title: {
		color: colors.white,
		fontSize: 24,
		fontWeight: "700"
	},
	bottom: {
		flex: 1,
		width: SCREEN_WIDTH,
		justifyContent: "center",
		padding: 20
	}
})

const mapStateToProps = state => {
	return {
		network: state.user.bitcoinNetwork
	}
}

const mapDispatchToProps = dispatch => {
	return bindActionCreators(
		{
			captureQr
		},
		dispatch
	)
}

export default connect(mapStateToProps, mapDispatchToProps)(ScanQrCode)