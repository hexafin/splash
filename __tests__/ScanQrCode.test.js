import "react-native";
import React from "react";
import { ScanQrCode } from "../src/components/ScanQrCode";

it("ScanQrCode renders correctly", () => {
	const snap = shallow(<ScanQrCode />);
	expect(snap).toMatchSnapshot();
});
