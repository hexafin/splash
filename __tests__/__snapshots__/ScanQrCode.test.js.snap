// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScanQrCode renders correctly 1`] = `
<_default
  bottomContent={
    <Component>
      <Component
        style={
          Object {
            "overflow": "hidden",
            "width": 750,
          }
        }
      >
        <Component
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../src/assets/images/headerWave.png",
            }
          }
          style={
            Object {
              "overflow": "hidden",
              "shadowOffset": Object {
                "height": 5,
                "width": 0,
              },
              "shadowOpacity": 0.12,
              "shadowRadius": 12,
              "top": -285,
              "width": 750,
            }
          }
        />
      </Component>
      <Component
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flexDirection": "row",
            "opacity": 0.75,
            "position": "absolute",
            "top": 65,
          }
        }
      >
        <Component
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 25,
              "paddingRight": 15,
            }
          }
        >
          Point at QR Code
        </Component>
        <Component
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../src/assets/icons/qrIcon.png",
            }
          }
          style={
            Object {
              "height": 30,
              "width": 30,
            }
          }
        />
      </Component>
      <AnimatedComponent
        style={
          Array [
            Object {
              "alignSelf": "center",
              "height": 270,
              "justifyContent": "space-between",
              "top": -275,
              "width": 270,
            },
            Object {
              "transform": Array [
                Object {
                  "scale": 0,
                },
              ],
            },
          ]
        }
      >
        <Component>
          <Component
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Component
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "height": 12,
                  "width": 50,
                }
              }
            />
            <Component
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "height": 12,
                  "width": 50,
                }
              }
            />
          </Component>
          <Component
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Component
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "height": 38,
                  "width": 12,
                }
              }
            />
            <Component
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "height": 38,
                  "width": 12,
                }
              }
            />
          </Component>
        </Component>
        <Component>
          <Component
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Component
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "height": 38,
                  "width": 12,
                }
              }
            />
            <Component
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "height": 38,
                  "width": 12,
                }
              }
            />
          </Component>
          <Component
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Component
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "height": 12,
                  "width": 50,
                }
              }
            />
            <Component
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "height": 12,
                  "width": 50,
                }
              }
            />
          </Component>
        </Component>
      </AnimatedComponent>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 49,
            "height": 98,
            "justifyContent": "center",
            "top": 359,
            "width": 98,
          }
        }
      >
        <Component
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../src/assets/icons/crossPrimary.png",
            }
          }
          style={
            Object {
              "height": 25,
              "width": 25,
            }
          }
        />
      </TouchableOpacity>
    </Component>
  }
  bottomViewStyle={
    Object {
      "position": "absolute",
    }
  }
  cameraStyle={
    Object {
      "height": 1334,
    }
  }
  containerStyle={
    Object {
      "backgroundColor": "#FFFFFF",
    }
  }
  onRead={[Function]}
/>
`;
